src/amo/components/HeroRecommendation/decorations.js

```diff
@@ -0,0 +1,96 @@
+  <svg
+    className="HeroRecommendation-decoration"
+    xmlns="http://www.w3.org/2000/svg"
+    viewBox="0 0 480 420"
+  >
+    <defs>
+      <linearGradient
+        id="background-noodle-1-a"
+        x1="37.554%"
+        x2="20.41%"
+        y1="51.425%"
+        y2="49.159%"
+      >
+        <stop offset="0%" stopColor="#592ACB" />
+        <stop offset="100%" stopColor="#312A65" />
+      </linearGradient>
+      <linearGradient
+        id="background-noodle-1-b"
+        x1="0%"
+        x2="100%"
+        y1="36.092%"
+        y2="36.092%"
+      >
+        <stop offset="0%" stopColor="#9059FF" />
+        <stop offset="100%" stopColor="#E31587" />
+      </linearGradient>
+    </defs>
+    <g fill="none" transform="matrix(-1 0 0 1 334 113)">
+      <path
+        fill="#312A65"
+        d="M334,307 L67,307 C119.112636,258.629663 161.220388,194.897267 187.226377,118.874319 C211.561794,47.736239 270.543021,3.04387234 334,0 L334,307 Z"
+      />
+      <path
+        fill="url(#background-noodle-1-a)"
+        d="M0,307 C17.5018261,297.066328 34.5740674,285.542356 51.0720591,272.398174 C121.246081,216.489776 216.932316,232.61482 271,307 L0,307 Z"
+      />
+      <path
+        fill="url(#background-noodle-1-b)"
+        d="M334,307 L114,307 C145.79039,282.134461 175.545243,252.742423 202.532663,218.864661 C237.964434,174.385973 285.119738,148.906295 334,143 L334,307 Z"
+      />
+    </g>
+  </svg>
+);
+
+export const svg2 = (
+  <svg
+    className="HeroRecommendation-decoration"
+    xmlns="http://www.w3.org/2000/svg"
+    viewBox="0 0 480 420"
+  >
+    <defs>
+      <linearGradient
+        id="background-noodle-2-a"
+        x1="43.792%"
+        x2="35.24%"
+        y1="72.633%"
+        y2="36.647%"
+      >
+        <stop offset="0%" stopColor="#592ACB" />
+        <stop offset="100%" stopColor="#312A65" />
+      </linearGradient>
+      <linearGradient
+        id="background-noodle-2-b"
+        x1="0%"
+        x2="100%"
+        y1="36.092%"
+        y2="36.092%"
+      >
+        <stop offset="0%" stopColor="#9059FF" />
+        <stop offset="100%" stopColor="#E31587" />
+      </linearGradient>
+    </defs>
+    <g fill="none" transform="translate(0 8)">
+      <path
+        fill="#312A65"
+        d="M0,412 L0,128.75365 C31.0094766,238.211493 89.1811504,337.098969 169.23274,412 L0,412 Z"
+      />
+      <path
+        fill="url(#background-noodle-2-a)"
+        d="M0,412 L0,0 C15.5111978,74.0180542 43.5322127,146.095994 84.6678403,212.277121 C139.500279,300.495165 211.038639,367.631123 290.979914,412 L0,412 Z"
+      />
+      <path
+        fill="url(#background-noodle-2-b)"
+        d="M0,412 L0,259.885196 C22.4809046,291.225016 47.4133204,320.994453 74.7800083,348.850262 C97.4362756,371.909451 121.070413,392.95665 145.512407,412 L0,412 Z"
+      />
+    </g>
+  </svg>
+);
+
+export const getDecoration = (randomizer: () => number = Math.random) => {
+  return randomizer() <= 0.5 ? svg1 : svg2;
```

Comment:
As you mentioned in the PR, this will cause the first client load to re-render as it will get a different random value. This is a no-go because it will affect performance of the home page. The first load of the home page cannot trigger a re-render.

One way to solve it is to put some value in Redux state so that when the client hydrates the server response, it will have a value to use.

Reply:
When I first looked at this patch, I thought I had seen this use case before and it took me some time to take a look and find the code.. Thanks @kumar303 for having looked at this patch in the meantime.

Here is the code: https://github.com/mozilla/addons-frontend/blob/master/src/core/reducers/heroBanners.js

This comes from an old version of the homepage. I guess we can reuse that.

Reply:
Thanks for the suggestions @kumar303 and @willdurand. I think I can actually solve this fairly easily once #8603 lands. I could set a value for the decoration during the `LOAD_HOME_DATA` action, which could then be used by the page. #8603 would ensure that `LOAD_HOME_DATA` would run whenever the home page is navigated to.